practice-one/src/Components/Common/Buttons.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// Mapping the props to corresponding Tailwind CSS classes
+const variantClasses = {
+  primary: 'bg-transparent hover:bg-primary hover:text-white focus:bg-primary focus:text-white',
+  secondary: 'bg-secondary text-white hover:bg-secondary hover:text-white focus:bg-secondary focus:text-white',
+  outline: 'bg-transparent hover:bg-primary hover:text-white hover:border-primary focus:bg-primary focus:text-white focus:border-primary',
+};
+
+const sizeClasses = {
+  small: 'px-3 py-1 text-sm',
+  medium: 'px-4 py-2 text-base',
+  large: 'px-6 py-3 text-lg',
+};
+
+const colorClasses = {
+  green: 'text-green-500',
+  blue: 'text-blue-500',
+  red: 'text-red-500',
+  black: 'text-black-500',
+  gray: 'text-gray-500',
+};
+// eslint-disable-next-line react/function-component-definition
+const Button = ({
+  text,
+  variant = 'primary',
+  size = 'medium',
+  rightIcon,
+  type,
+  color,
+  onClick = () => {},
+}) => {
+  // Define the Tailwind CSS classes based on the props
+  const buttonClasses = `inline-flex items-center justify-center rounded ${sizeClasses[size]} ${variantClasses[variant]} ${colorClasses[color]}`;
+
+  return (
+    <button
+      type={Boolean ? 'button' : 'submit'}

```

Comment:
what is this?
Did you understand the prop-type

```js
type: PropTypes.oneOf(['button', 'submit', 'reset']),
```

`type` is one of ['button', 'submit', 'reset'] why have "boolean" here

Reply:
What is "Boolean"? does it always return true? where is the "reset" value?

Reply:
Done